### PR TITLE
chore(MenuButton): remove   showTooltipOnlyOnTriggerElement

### PR DESCRIPTION
**Pull request by talkor**

### CODE REVIEW

File: packages/core/src/components/MenuButton/__tests__/__snapshots__/MenuButton.test.tsx.snap [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
 
 exports[`MenuButton renders correctly with .dummy-class-name 1`] = `
-<button
-  aria-disabled={false}
-  aria-expanded={false}
-  aria-haspopup="true"
-  aria-label="Menu"
-  className="wrapper dummy-class-name sizeSmall"
-  data-testid="menu-button"
+<span
```
Comment (talkor) [inline packages/core/src/components/MenuButton/__tests__/__snapshots__/MenuButton.test.tsx.snap:4]: @YossiSaadi I'm not sure about this change, is it ok a11y wise? Do you remember?

File: packages/core/src/components/MenuButton/MenuButton.tsx [hunk 2]
```diff
@@ -366,10 +361,7 @@ const MenuButton: VibeComponent<MenuButtonProps> & {
       </Tooltip>
     );
 
-    if (showTooltipOnlyOnTriggerElement) {
-      return dialogNode(tooltipNode(triggerElementNode));
-    }
-    return tooltipNode(dialogNode(triggerElementNode));
+    return dialogNode(tooltipNode(triggerElementNode));
```
Comment (YossiSaadi) [inline packages/core/src/components/MenuButton/MenuButton.tsx:364]: you can inline it I guess and remove the consts, if I remember correctly I've only added them for `showTooltipOnlyOnTriggerElement` functionality

File: packages/core/src/components/MenuButton/__tests__/__snapshots__/MenuButton.test.tsx.snap (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/src/components/MenuButton/__tests__/__snapshots__/MenuButton.test.tsx.snap:4]: why did it happen?
I don't think there's an a11y issue here, but I'm not sure that it should have happened
can you compare to the snapshot before the `showTooltipOnlyOnTriggerElement` change? if the snapshot looked exactly as it is right now?

File: packages/codemod/transformations/core/v2-to-v3/tooltip-component-migration.ts [hunk 3]
```diff
@@ -20,6 +21,7 @@ function transform({ j, root }: TransformationContext) {
 
   elements.forEach(elementPath => {
     removeProp(j, elementPath, "withMaxWidth");
+    removeProp(j, elementPath, "showTooltipOnlyOnTriggerElement");
```
Comment (YossiSaadi) [inline packages/codemod/transformations/core/v2-to-v3/tooltip-component-migration.ts:24]:
```suggestion
    removeProp(j, elementPath, "withMaxWidth", "showTooltipOnlyOnTriggerElement");
```

should also work, should save an iteration of `O(n^2)` (probably negligible, so for your consideration)

**Review by YossiSaadi — APPROVED** [review]
Looks cool! not sure about the snapshot, we should compare the snapshots before the `"showTooltipOnlyOnTriggerElement"` first introduced

**Pull request closed without merging**